libkbfs/favorites.go

```diff
@@ -107,6 +112,87 @@ func NewFavorites(config Config) *Favorites {
 	return newFavoritesWithChan(config, make(chan *favReq, 100))
 }
 
+func (f *Favorites) readCacheFromDisk(ctx context.Context) error {
+	// Read the encrypted cache from disk
+	folder := f.config.StorageRoot()
+	subfolder := "kbfs.favorites.cache"
+	session, err := f.config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	user := string(session.UID)
+	path := filepath.Join(folder, subfolder, user)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	// Send the data to the service to be decrypted
+	decryptedData, err := f.config.KeybaseService().DecryptFavorites(ctx, data)
+	if err != nil {
+		return err
+	}
+
+	// Decode the data into the a map
+	decoder := gob.NewDecoder(bytes.NewBuffer(decryptedData))
+	var cache map[Favorite]bool
+	err = decoder.Decode(&cache)
+	if err != nil {
+		return err
+	}
+
+	f.cache = cache
+	return nil
+}
+
+func (f *Favorites) writeCacheToDisk(ctx context.Context) error {
+	// Encode the cache map into a byte buffer
+	buf := &bytes.Buffer{}
+	encoder := gob.NewEncoder(buf)
+	err := encoder.Encode(f.cache)
+	if err != nil {
+		return err
+	}
+
+	// Send the byte buffer to the service for encryption
+	data, err := f.config.KeybaseService().EncryptFavorites(ctx, buf.Bytes())
+	if err != nil {
+		return err
+	}
+
+	// Write the encrypted cache to disk
+	folder := f.config.StorageRoot()
+	subfolder := "kbfs.favorites.cache"
+	session, err := f.config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	user := string(session.UID)
+	path := filepath.Join(folder, subfolder, user)
```

Comment:
Just realized that this path has to have the UID in there somewhere.  Multiple Keybase accounts should be able to cache their favorites to disk.

Reply:
Nevermind, I'm dumb.